servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/SslConfig.java

```diff
@@ -186,4 +197,19 @@ default Duration handshakeTimeout() {   // FIXME 0.43 - remove default implement
     default int maxCertificateListBytes() {
         return 0;
     }
+
+    /**
+     * Defines filtering logic for {@link #ciphers() ciphers suites}.
+     */
+    enum CipherSuiteFilter {
+        /**
+         * Will take all requested ciphers suites as-is without any filtering.
+         */
+        IDENTITY,
+
+        /**
+         * Will filter all requested ciphers suites out that are not supported by the current {@link SSLEngine}.
+         */
+        SUPPORTED
+    }
```

Comment:
I think this is nice and simple, simpler than the Netty implementation, but do we want to lock ourselves out of doing something more generic in the future without breaking API? Like Netty, we can just provide these two options as default.

Reply:
Based on Netty experience, there is no really a need to implement any other filtering strategies besides these 2 options. I picked enum instead of just a boolean to leave us a room to add more strategies later. 

If there is a well-defined need to provide the full filter API in the future, it can be additive to this enum. Some users will keep using the enum, others may opt-in for more advance API. It's just hard to come up with a use-case that requires something more than these 2.

Reply:
Can you give an example of what you mean by additive? I interpret that is there being a parallel method, something like `CipherSuiteFilterFunction cipherSuiteFilterV2()` (perhaps more clever names but of this form) which seems unpleasant.

Reply:
Correct, it will be another method on `SslConfig` that returns a filtering function instead of an enum, and we will deprecate the pre-existing method. The enum will be able to stay and work as a holder for default filter implementations with an extra method that returns a function. 

I'm agreed that it will be unpleasant situation, but the likelihood of that situation is quite low. I can not imagine other way to filter ciphers except these two. Even if there will a 3rd way in the future, we can just add a new enum value instead of exposing a function.

Reply:
Sticking with an enum makes sense so long as we're quite certain we'll never need a custom filter. 👍